src/main/kotlin/open/tresorier/services/OperationService.kt

```diff
@@ -12,19 +13,23 @@ import java.io.File
 import java.io.BufferedReader
 import open.tresorier.model.enum.ActionEnum
 
-class OperationService(private val operationDao: IOperationDao, private val authorizationService: AuthorizationService, private val userActivityService: UserActivityService) {
+class OperationService(
+    private val operationDao: IOperationDao,
+    private val authorizationService: AuthorizationService,
+    private val accountDao: IAccountDao,
+    private val userActivityService: UserActivityService) {
 
     fun createInitialOperation(person: Person, account: Account, day: Day, amount: Int){
         authorizationService.cancelIfUserIsUnauthorized(person, account)
-        val operation = Operation(account.id, day, Category.INCOME_ID, amount, 1, "Montant initial", false, false)
+        val operation = Operation(account.id, day, Category.INCOME_ID, amount, 1, "Montant initial", false, false, null)
         operationDao.insert(operation)
         authorizationService.cancelIfUserIsUnauthorized(person, operation)
     }
 
-    fun create(person: Person, account: Account, day: Day, category: Category?, amount: Int?, memo: String?, pending: Boolean?) : Operation {
+    fun create(person: Person, account: Account, day: Day, category: Category?, amount: Int?, memo: String?, pending: Boolean?, motherOperationId: String?) : Operation {
```

Comment:
As a rule, to ensure that the mother operation does exist, you should give the operation instead of the operationId in parameters of this function

Reply:
done